docs/user/configuring-agent.md

```diff
@@ -135,6 +136,117 @@ status:
+
+### TPM Configuration Parameters
+
+The `tpm` configuration object accepts the following parameters:
+
+| Parameter | Type | Required | Description |
+| --------- | ---- | :------: | ----------- |
+| `enabled` | `boolean` | | Enable TPM device identity. When true, the agent uses TPM for key generation and protection. Default: `false` |
+| `device-path` | `string` | | Path to the TPM device. If not specified, the agent auto-discovers available TPM devices, preferring resource manager devices (`/dev/tpmrm*`) over direct devices. Default: auto-discovery |
+| `auth-enabled` | `boolean` | | Enable TPM owner hierarchy password authentication. Should only be used in development/test environments. Default: `false` |
+| `storage-file-path` | `string` | | File path for TPM key handle persistence. Default: `/var/lib/flightctl/tpm-blob.yaml` |
+
+### Example TPM Configuration
+
+```yaml
+# /etc/flightctl/config.yaml
+[...]
+tpm:
+  enabled: true
+  device-path: /dev/tpm0
+  auth-enabled: false
+  storage-file-path: /var/lib/flightctl/tpm-blob.yaml
+
+spec-fetch-interval: 60s
+status-update-interval: 60s
+```
+
+### TPM Auto-Discovery
+
+When `device-path` is not specified or is empty, the agent automatically discovers TPM devices by:
+
+1. Scanning `/sys/class/tpm/` for available TPM devices
+2. Validating TPM version (must be 2.0)
+3. Preferring resource manager devices (`/dev/tpmrm*`) over direct devices (`/dev/tpm*`)
+4. Using the first valid TPM 2.0 device found
+
+### TPM Requirements
+
+* **Hardware**: TPM 2.0 compliant module (TPM 1.2 is not supported)
+* **Kernel**: Linux kernel with TPM 2.0 support
+* **Permissions**: Agent must have read/write access to TPM device
+* **CA Certificates**: TPM manufacturer CA certificates must be installed on the Flight Control service
+
+For detailed information about TPM authentication architecture and certificate requirements, see [TPM Device Authentication](tpm-authentication.md).
+
+### TPM Troubleshooting
+
+#### TPM Not Detected
+
+If the agent cannot detect the TPM device:
+
+```bash
+# Check if TPM device exists
+ls -la /dev/tpm* /dev/tpmrm*
+
+# Verify TPM version (should output: 2)
+cat /sys/class/tpm/tpm0/tpm_version_major
+
+# Check device permissions
+ls -l /dev/tpm0
+# Agent user should have read/write access
+```
+
+#### TPM Initialization Errors
+
+Check agent logs for TPM-related errors:
+
+```bash
+# View agent logs
+journalctl -u flightctl-agent -f
+
+# Filter for TPM messages
+journalctl -u flightctl-agent | grep -i tpm
+```
+
+Enable debug logging for detailed TPM information:
+
+```yaml
+# /etc/flightctl/config.yaml
+log-level: debug  # or trace for maximum detail
+```
+
+#### TPM Owner Password
+
+If the TPM requires owner hierarchy authentication:
+
+```yaml
+tpm:
+  enabled: true
+  auth-enabled: true  # Use well-known password for TPM ownership
```

Comment:
This generates a random password that survives reboots, but if the storage file is lost, so is our ability to use the ownership hierarchy and a tpm2_clear must be issued.